assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -145,74 +129,66 @@ export class SpreadsheetFrontend extends Component {
-      this.state.spreadSheetData.header :
-      placeholderData.header;
-
-    return (
-      <Fragment>
-        <section className={`block block-spreadsheet ${this.props.className ?? ''}`}>
-          <input className="spreadsheet-search form-control"
-            type="text"
-            value={this.state.searchText}
-            onChange={event => this.setState({searchText: event.target.value})}
-            placeholder={__('Search data', 'planet4-blocks')}
-          />
-          <div className="table-wrapper">
-            <table className={`spreadsheet-table is-color-${this.props.color ?? 'grey'}`}>
-              <thead>
-                <tr>
-                  {
-                    headers.map((cell, index) => (
-                      <th
-                        className={(
-                          index === this.state.sortColumnIndex ?
-                            `spreadsheet-sorted-by sort-${this.state.sortDirection}` :
-                            ''
-                        )}
-                        onClick={() => {
-                          this.onHeaderClick(index);
-                        }}
-                        key={index}
-                        scope="col"
-                        title={cell}>
-                        <button>
-                          { cell }
-                          <ArrowIcon />
-                        </button>
-                      </th>
-                    ))
-                  }
-                </tr>
-              </thead>
-              <tbody>
-                {
-                  this.state.loading === true ?
-                    <tr>
-                      <td colSpan="99999">
-                        <div className="spreadsheet-loading">{ __('Loading spreadsheet data…', 'planet4-blocks') }</div>
-                      </td>
-                    </tr> :
-                    this.renderRows()
-                }
-              </tbody>
-            </table>
-          </div>
-        </section>
-      </Fragment>
-    );
-  }
-}
+  };
+
+  const headers = spreadsheetData ? spreadsheetData.header : placeholderData.header;
+
+  return (
+    <section className={`block block-spreadsheet ${className ?? ''}`}>
+      <input className="spreadsheet-search form-control"
+        type="text"
+        value={searchText}
+        onChange={event => setSearchText(event.target.value)}
+        placeholder={__('Search data', 'planet4-blocks')}
+      />
+      <div className="table-wrapper">
+        <table className={`spreadsheet-table is-color-${color ?? 'grey'}`}>
+          <thead>
+            <tr>
+              {
+                headers.map((cell, index) => (
+                  <th
+                    className={(
+                      index === sortColumnIndex ?
+                        `spreadsheet-sorted-by sort-${sortDirection}` :
+                        ''
+                    )}
+                    onClick={() => onHeaderClick(index)}
```

Comment:
What do you think about the below suggestion?

Pass the `index` value through the dataset
```
...
data-index={index}
onClick={onHeaderClick}
...
```
Then from [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/adac09d072079130e1bb7c68c5c12331ddbe2a92/assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js#L31-L38) we would do something like this:
```js
  const onHeaderClick = (evt) => {
   const index = evt.currentTarget.dataset.index;
    if (index === sortColumnIndex) {
      setSortDirection(sortDirection === 'asc' ? 'desc' : 'asc');
    } else {
      setSortColumnIndex(index);
      setSortDirection('asc');
    }
  };
```

What will be gain? To avoid create multiple `anonymus` function through the render method and use only one.